### PR TITLE
CFINSPEC-580 Dependent profiles fix for any scheme of version used in profiles

### LINKED ISSUE #6409: Dependent profiles are not found when profile version does not follow semver
In InSpec 5.21.29 dependent profiles specified by path that do not have version failed to run with the following error

```
ERROR: Failed to load profile vmware-vcsa-7.0-stig-baseline: Failed to load source for controls/photon.rb: Cannot load 'photon' since it is not listed as a dependency of vmware-vcsa-7.0-stig-baseline.

Dependencies available from this context are:
    eam-1.0.4
    lookup-1.0.4
    perfcharts-1.0.4
    photon-3.0 Release 1 Version 5
    postgres-1.0.4
    rhttpproxy-1.0.4
    sts-1.0.4
    vami-1.0.4
    vsphere-ui-1.0.4
```

# Describe the problem
Profiles with dependent profiles should run those profiles without having to specify a version.

inspec.yml depends block for the profile generating the error that worked in prior versions without issue.

```
depends:
  - name: eam
    path: eam
  - name: lookup
    path: lookup
  - name: perfcharts
    path: perfcharts
  - name: photon
    path: ../vmware-photon-3.0-stig-inspec-baseline
  - name: postgres
    path: postgresql
  - name: rhttpproxy
    path: rhttpproxy
  - name: sts
    path: sts
  - name: vami
    path: vami
  - name: vsphere-ui
    path: ui
```

Possibly related to https://github.com/inspec/inspec/pull/6377

### PR DESCRIPTION
**Pull request by Nik08**
Signed-off-by: Nikita Mathur <nikita.mathur@chef.io>

## Description

Fix to load dependent profiles which follow any scheme for versioning profiles. Before this, it was only restricted to semver versioning.

## Related Issue

Fixes #6409 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/dsl.rb [hunk 1]
```diff
@@ -91,16 +91,12 @@ def self.load_spec_files_for_profile(bind_context, opts, &block)
...
-      dependencies.list.keys.each do |key|
+      dependencies.list.each do |key, value|
         # 1. Fetching VERSION from a profile dependency name which is in a format NAME-VERSION.
         # 2. Matching original profile dependency name with profile name used with include or require control DSL.
-        fetching_semver = key.match(regex_for_semver).to_s
-        unless fetching_semver.nil? || fetching_semver.empty?
-          profile_id_key = key.split("-#{fetching_semver}")[0]
-          new_profile_id = key if profile_id_key == profile_id
-        end
+        source_version = value.source_version
```
Comment (Vasu1105) [inline lib/inspec/dsl.rb:97]: in case value.source_version is empty or nil. Are we covering that case here?

Comment (Nik08) [reply lib/inspec/dsl.rb:97]: I will cover it up.

**Review by Vasu1105 — APPROVED** [review]
I only had a doubt about using blank but that got clarified by @Nik08. So this looks good to me.

**Comment by bdwyertech** [comment]
This introduces a dependency on ActiveSupport to use the blank method.  I am receiving errors and had to revert to 5.21.29 to resolve.

@Nik08 @Vasu1105 -- can we please get rid of this or add active_support to the gem dependencies?

**Comment by bdwyertech** [comment]
> I only had a doubt about using blank but that got clarified by @Nik08. So this looks good to me.

You were correct in questioning usage of blank.

**Comment by Nik08** [comment]
@bdwyertech Inspec uses train which has a listed gem dependency on `active-support` https://github.com/inspec/train/blob/main/train.gemspec#L34.
And this has been used in inspec for over last 2 years now. There is the usage of `blank?` in a lot of instances within InSpec.

Maybe resolving your dependency on [train](https://github.com/inspec/train), would help with this.

**Pull request merged**